error undefined reference to `_mm512_reduce_add_pd'

Hello

I have ran into the following error while building openBLAS 0.3.19 on RHEL7 with gcc-6.3 and DYNAMIC_ARCH=1

../libopenblas_nonthreaded-r0.3.19.a(dgemm_small_kernel_nn_SKYLAKEX.o): In function `dgemm_small_kernel_nn_SKYLAKEX':
dgemm_small_kernel_nn_skylakex.c:(.text+0x2ddd): undefined reference to `_mm512_reduce_add_pd'
dgemm_small_kernel_nn_skylakex.c:(.text+0x2e1c): undefined reference to `_mm512_reduce_add_pd'
dgemm_small_kernel_nn_skylakex.c:(.text+0x2e8e): undefined reference to `_mm512_reduce_add_pd'
dgemm_small_kernel_nn_skylakex.c:(.text+0x2eea): undefined reference to `_mm512_reduce_add_pd'
dgemm_small_kernel_nn_skylakex.c:(.text+0x30c7): undefined reference to `_mm512_reduce_add_pd'
../libopenblas_nonthreaded-r0.3.19.a(dgemm_small_kernel_nn_SKYLAKEX.o):dgemm_small_kernel_nn_skylakex.c:(.text+0x3106): more undefined references to `_mm512_reduce_add_pd' follow
../libopenblas_nonthreaded-r0.3.19.a(sgemm_small_kernel_nn_SKYLAKEX.o): In function `sgemm_small_kernel_nn_SKYLAKEX':
sgemm_small_kernel_nn_skylakex.c:(.text+0x2ca3): undefined reference to `_mm512_reduce_add_ps'
sgemm_small_kernel_nn_skylakex.c:(.text+0x2cdc): undefined reference to `_mm512_reduce_add_ps'
sgemm_small_kernel_nn_skylakex.c:(.text+0x2d42): undefined reference to `_mm512_reduce_add_ps'
sgemm_small_kernel_nn_skylakex.c:(.text+0x2d95): undefined reference to `_mm512_reduce_add_ps'
sgemm_small_kernel_nn_skylakex.c:(.text+0x2f25): undefined reference to `_mm512_reduce_add_ps'
../libopenblas_nonthreaded-r0.3.19.a(sgemm_small_kernel_nn_SKYLAKEX.o):sgemm_small_kernel_nn_skylakex.c:(.text+0x2f5e): more undefined references to `_mm512_reduce_add_ps' follow
collect2: error: ld returned 1 exit status

any ideas how to get around this issue?

Regards, Roman

--- a/kernel/x86_64/dgemm_small_kernel_nn_skylakex.c
+++ b/kernel/x86_64/dgemm_small_kernel_nn_skylakex.c
@@ -24,6 +24,7 @@ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *****************************************************************************/
+#if (( defined(__GNUC__)  && __GNUC__   > 6 && defined(__AVX512CD__)) || (defined(__clang__) && __clang_major__ >= 9))
 
 #include <immintrin.h>
 #include "common.h"
@@ -588,3 +589,7 @@ int CNAME(BLASLONG M, BLASLONG N, BLASLONG K, FLOAT * A, BLASLONG lda, FLOAT alp
 	}
 	return 0;
 }
+#else
+#include ../generic/gemm_small_matrix_kernel_nn.c
+#endif
+

--- a/kernel/x86_64/dgemm_small_kernel_tn_skylakex.c
+++ b/kernel/x86_64/dgemm_small_kernel_tn_skylakex.c
@@ -24,6 +24,7 @@ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *****************************************************************************/
+#if (( defined(__GNUC__)  && __GNUC__   > 6 && defined(__AVX512CD__)) || (defined(__clang__) && __clang_major__ >= 9))
 
 #include <immintrin.h>
 #include "common.h"
@@ -320,3 +321,7 @@ int CNAME(BLASLONG M, BLASLONG N, BLASLONG K, FLOAT * A, BLASLONG lda, FLOAT alp
 	}
 	return 0;
 }
+#else
+#include ../generic/gemm_small_matrix_kernel_tn.c
+#endif
+

--- a/kernel/x86_64/sgemm_small_kernel_nn_skylakex.c
+++ b/kernel/x86_64/sgemm_small_kernel_nn_skylakex.c
@@ -24,6 +24,7 @@ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *****************************************************************************/
+#if (( defined(__GNUC__)  && __GNUC__   > 6 && defined(__AVX512CD__)) || (defined(__clang__) && __clang_major__ >= 9))
 
 #include <immintrin.h>
 #include "common.h"
@@ -610,3 +611,7 @@ int CNAME(BLASLONG M, BLASLONG N, BLASLONG K, FLOAT * A, BLASLONG lda, FLOAT alp
 	}
 	return 0;
 }
+#else
+#include ../generic/gemm_small_matrix_kernel_nn.c
+#endif
+

--- a/kernel/x86_64/sgemm_small_kernel_tn_skylakex.c
+++ b/kernel/x86_64/sgemm_small_kernel_tn_skylakex.c
@@ -24,6 +24,7 @@ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *****************************************************************************/
+#if (( defined(__GNUC__)  && __GNUC__   > 6 && defined(__AVX512CD__)) || (defined(__clang__) && __clang_major__ >= 9))
 
 #include <immintrin.h>
 #include "common.h"
@@ -314,3 +315,7 @@ int CNAME(BLASLONG M, BLASLONG N, BLASLONG K, FLOAT * A, BLASLONG lda, FLOAT alp
 	}
 	return 0;
 }
+#else
+#include ../generic/gemm_small_matrix_kernel_tn.c
+#endif
+

